### PR TITLE
AB#39541: Validate Organisation To Check For Default Aggregation Values

### PR DESCRIPTION
**Pull request by Flemingjp**
For a collection to be properly aggregated, the organsiation must set up the default aggregation values for `AccessCondition` and `CollectionType`. This is configured in the Directory UI under the Bulk Submissions settings.

- Users cannot generate API credentials until default values are configured
- Aggregator will skip aggregating a collection if the Organisation does not have configured values

### CODE REVIEW

File: src/Directory/Biobanks.Web/Views/Biobank/Submissions.cshtml [hunk 1]
```diff
@@ -2,162 +2,174 @@
...
-                class="btn btn-success pull-right">
-            Generate New Client Secret
-        </button>
-    </div>
-</div>
+@* API Credentials *@
+<h3>Submission API Authentication</h3>
 
+@* API Client Credential Only Accessible Once Organisation Defaults Are Set *@
+@if (Model.AccessCondition != null && Model.CollectionType != null)
```
Comment (Flemingjp) [inline src/Directory/Biobanks.Web/Views/Biobank/Submissions.cshtml:107]: This section is now conditional

File: src/Core/Jobs/AggregatorJob.cs [hunk 2]
```diff
@@ -84,9 +84,19 @@ private async Task AggregateCollectionSamples(LiveSample baseSample, IEnumerable
                 throw new ArgumentException("Supplied baseSample is null, or is referencing a non-existent SampleContent");
 
             var organisation = await _organisationService.GetById(baseSample.OrganisationId);
+            var collectionName = _aggregationService.GenerateCollectionName(baseSample);
+
+            // Check Organisation Has Properly Configured Default Values
+            if (organisation.AccessCondition is null || organisation.CollectionType is null)
+            {
+                // Log Warning
+                Console.WriteLine($"Could not aggregate {collectionName} for {organisation.Name} due to missing default aggregation values.");
```
Comment (Flemingjp) [inline src/Core/Jobs/AggregatorJob.cs:93]: Not sure on the best way to expose a warning via Hangfire. Looking around resulted in an extension to Hangfire https://github.com/pieceofsummer/Hangfire.Console

Comment (beforan) [reply src/Core/Jobs/AggregatorJob.cs:93]: we need to pursue the logging plugin, but that's kind of its own pbi.

This behaviour increases its priority though, i would say.

Comment (Flemingjp) [reply src/Core/Jobs/AggregatorJob.cs:93]: OK I would suggest keeping it as this for time being, but with a future aim to implement Hangfire logging

**Review by JosephLavagna — APPROVED** [review]

**Review by blargly — APPROVED** [review]

**Pull request merged**